src/main/scala/cromwell/binding/Task.scala

```diff
@@ -1,7 +1,45 @@
 package cromwell.binding
 
-import cromwell.binding.command.Command
-import cromwell.binding.types.WdlType
+import cromwell.binding.command.{ParameterCommandPart, StringCommandPart, Command}
+import cromwell.binding.types._
+import cromwell.parser.AstTools.AstNodeName
+import cromwell.parser.WdlParser._
+import cromwell.parser.AstTools.{EnhancedAstNode, EnhancedAstSeq}
+import scala.collection.JavaConverters._
+import scala.language.postfixOps
+
+object Task {
+  def apply(ast: Ast, wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Task = {
+    val name = ast.getAttribute("name").asInstanceOf[Terminal].getSourceString
+
+    val dupeParamAsts = ast.findAsts(AstNodeName.CommandParameter).duplicatesByName
+    if (dupeParamAsts.nonEmpty) {
+      throw new SyntaxError(wdlSyntaxErrorFormatter.taskHasDuplicatedInputs(ast, dupeParamAsts))
+    }
+
+    val commandAsts = ast.findAsts(AstNodeName.Command)
+    if (commandAsts.size != 1) throw new UnsupportedOperationException("Expecting only one Command AST")
+    val command = Command(commandAsts.head)
+    val outputs = ast.findAsts(AstNodeName.Output) map {TaskOutput(_)}
+    new Task(name, command, outputs, buildRuntimeAttributes(ast), ast)
+  }
+
+  // TODO/FIXME: If RuntimeAttributes turned into a real type (i.e. case class) the following crap could go into its construction
+  private def buildRuntimeAttributes(ast: Ast): RuntimeAttributes = {
+    val asts = ast.findAsts(AstNodeName.Runtime)
+    if (asts.size > 1) throw new UnsupportedOperationException("Only one runtime block may be defined per task")
+    val astList = asts.headOption map {_.getAttribute("map").asInstanceOf[AstList]}
```

Comment:
In general I wish we had a codified way to handle this sort of thing as it happens throughout the WdlNamespace family of stuff and while it is crystal clear once you look at it, it's easy to get confused